docs/src/references/c8y-configuration-management.md

```diff
@@ -163,3 +278,229 @@ Note that:
 * Since the type of configuration file is used as an MQTT topic name, the characters `#` and `+` cannot be used in a type name.
   If such a character is used in a type name (or in the path of a configuration file without explicit type),
   then the whole plugin configuration `/etc/tedge/c8y/c8y-configuration-plugin.toml` is considered ill-formed.
+
+## Configuration protocol between thin-edge and the child-devices
+
+The configuration plugin `c8y_configuration_plugin` can act as a proxy between the cloud and a child-device.
+However, for that to work, a client must be installed on the child device
+to perform the actual configuration updates pushed by thin-edge on behalf of the cloud.
+While the configuration plugin tells what need to be updated and when,
+only the child device specific client can control where and how these updates can be applied.
+
+* The responsibility of the configuration plugin is to
+  * interact with the cloud, receiving the configuration updates and requests,
+  * publish and collect configuration files via an HTTP-based file transfer service over the local network,
+  * notify the child devices via MQTT when configuration files are to be updated or requested from the cloud,
+  * listen for child devices MQTT configuration status messages.
+* The child-device specific software is an MQTT+HTTP client that
+  * interact with the child-device system, updating and requesting the target configuration files
```

Comment:
The child-device will never request a configuration file. Maybe it downloads a file on demand.

Reply:
I will improve the working since I was not speaking about the child device requesting a configuration to the cloud or thin-edge. But the child-device configuration software that read / access the actual configuration on the child device.